.github/workflows/fast-forward.yml

```diff
@@ -0,0 +1,45 @@
+name: Fast-forward tracking branch for latest CodeQL version
+on:
+  workflow_dispatch:
+
+jobs:
+  fast-forward:
+    name: Fast-forward tracking branch for latest CodeQL version
+    runs-on: ubuntu-latest
+    if: github.repository == 'github/codeql'
+    permissions:
+      contents: write
+    env:
+      BRANCH_NAME: 'lgtm.com'
+    steps:
+      - name: Validate chosen branch
+        if: ${{ !startsWith(github.ref_name, 'codeql-cli-') }}
+        shell: bash
```

Comment:
Minor, optional: I believe the `${{ }}` can be dropped for the `if` property, and `shell: bash` is also redundant in a workflow.

Reply:
Apparently not when you have a `!` operator! At least the YAML editor is unhappy with me if I do that, though Actions may accept the syntax.

Reply:
Ah!  Bangs are special in YAML so you'd need to quote the string.  I'd expect `if: '!startsWith(github.ref_name, 'codeql-cli-')'` should work, but it's not really better than the `${{ }}`.